cmd/basin/publication.go

```diff
@@ -463,6 +515,62 @@ func parsePublicationName(name string) (ns string, rel string, err error) {
 	return
 }
 
+func inspectTable(ctx context.Context, tx pgx.Tx, rel string) ([]app.Column, error) {
+	rows, err := tx.Query(ctx,
+		`
+		WITH primary_key_info AS
+			(SELECT tc.constraint_schema,
+					tc.table_name,
+					ccu.column_name
+			FROM information_schema.table_constraints tc
+			JOIN information_schema.constraint_column_usage AS ccu USING (CONSTRAINT_SCHEMA, CONSTRAINT_NAME)
+			WHERE constraint_type = 'PRIMARY KEY' )
+		SELECT
+			c.column_name,
+			c.data_type,
+			c.is_nullable = 'YES' AS is_nullable,
+			pki.column_name IS NOT NULL AS is_primary
+		FROM information_schema.columns AS c
+		LEFT JOIN primary_key_info pki ON c.table_schema = pki.constraint_schema
+			AND pki.table_name = c.table_name
+			AND pki.column_name = c.column_name
+		WHERE c.table_name = $1; 
+		`, rel,
+	)
+	if err != nil {
+		return []app.Column{}, fmt.Errorf("failed to fetch schema")
+	}
+	defer rows.Close()
+
+	var colName, typ string
+	var isNull, isPrimary bool
+	var columns []app.Column
+	for rows.Next() {
+		if err := rows.Scan(&colName, &typ, &isNull, &isPrimary); err != nil {
+			return []app.Column{}, fmt.Errorf("scan: %s", err)
+		}
+
+		columns = append(columns, app.Column{
+			Name:      colName,
+			Typ:       typ,
+			IsNull:    isNull,
+			IsPrimary: isPrimary,
+		})
+	}
+	return columns, nil
+}
+
+func mkDBDir(dir, pub string) error {
+	if err := os.Mkdir(path.Join(dir, pub), 0o755); err != nil {
```

Comment:
Maybe better using [MkdirAll](https://pkg.go.dev/os#MkdirAll) so it ignores if the dir already exists? 

> MkdirAll creates a directory named path, along with any necessary parents, and returns nil, or else returns an error. The permission bits perm (before umask) are used for all directories that MkdirAll creates. If path is already a directory, MkdirAll does nothing and returns nil.